### PR TITLE
Deps update

### PR DESCRIPTION
**Pull request by hazboun**
I managed to update most of the dependencies to latest versions, including sails, webpack, react, babel, and others.

### CODE REVIEW

**Comment by markmur** [comment]
😆 That's a lot of changes.

Thanks @hazboun, I'll take a look at this sometime this week 👍

**Comment by hazboun** [comment]
No problem @markmur :)

File: template/.babelrc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "plugins": ["transform-runtime", "react-html-attrs"],
-  "presets": ["es2015", "react"]
+  "plugins": ["react-hot-loader/babel", "transform-runtime", "react-html-attrs"],
+  "presets": ["react", "es2015"]
```
Comment (markmur) [inline template/.babelrc:3]: I should probably be using `babel-preset-env` here... It can be updated in another PR.

File: template/api/responses/badRequest.js [hunk 2]
```diff
@@ -1,76 +0,0 @@
-/**
```
Comment (markmur) [inline template/api/responses/badRequest.js:1]: I'm guessing this stuff is included in Sails v1.0 now?

File: template/package.json [hunk 3]
```diff
@@ -52,27 +52,31 @@
...
     "enzyme": "^2.4.1",
+    "html-webpack-plugin": "^2.30.1",
     "mocha": "^3.0.2",
-    "node-sass": "^3.4.2",
+    "node-sass": "^4.9.0",
     "precss": "^1.4.0",
     "react-addons-test-utils": "^15.0.1",
-    "react-hot-loader": "^1.3.0",
+    "react-hot-loader": "^4.2.0",
     "sails-hook-autoreload": "^0.13.1",
```
Comment (markmur) [inline template/package.json:75]: Does this still work with Sails v1?

File: template/package.json [hunk 4]
```diff
@@ -14,8 +14,8 @@
     "boilerplate"
   ],
   "dependencies": {
+    "@sailshq/lodash": "^3.10.3",
```
Comment (markmur) [inline template/package.json:17]: Why is lodash this scoped to @sailshq? Do they use a fork?

File: template/package.json [hunk 5]
```diff
@@ -52,27 +52,31 @@
...
-    "react-hot-loader": "^1.3.0",
+    "react-hot-loader": "^4.2.0",
     "sails-hook-autoreload": "^0.13.1",
-    "sass-loader": "^3.2.0",
+    "sass-loader": "^7.0.1",
     "sinon": "^1.17.5",
-    "style-loader": "^0.13.1",
-    "webpack-dev-server": "^1.14.1"
+    "style-loader": "^0.21.0",
+    "webpack": "^3.10.0",
```
Comment (markmur) [inline template/package.json]: Is there any reason you're not using `webpack@4`?

File: template/webpack.config.prod.js [hunk 6]
```diff
@@ -24,34 +20,32 @@ module.exports = {
...
-    root: path.join(__dirname, ''),
-    modulesDirectories: [
-      'web_modules',
-      'node_modules',
-      'assets',
-      'assets/components',
-    ],
-    extensions: ['', '.webpack.js', '.web.js', '.js', '.jsx'],
+    modules: [__dirname, 'node_modules'],
+        alias:{
```
Comment (markmur) [inline template/webpack.config.prod.js:36]: (pedantic) extract tab here before `alias`

**Review by markmur — CHANGES_REQUESTED** [review]
Thanks for the PR! It all looks good but I have a couple of small questions before approving.

I'd also prefer if it used the latest version of `webpack` (`4.8.3`), unless there's some reason why you chose to use v3?

File: template/api/responses/badRequest.js (hunk 2 above)
Comment (hazboun) [reply template/api/responses/badRequest.js:1]: right, that's why I had to remove them.

File: template/package.json (hunk 3 above)
Comment (hazboun) [reply template/package.json:75]: yes it does, but it only works with webpack V3 and that's why I couldn't use latest version of webpack.

File: template/package.json (hunk 4 above)
Comment (hazboun) [reply template/package.json:17]: seems so, I generated a new Sailsjs project with the latest version and noticed they're using the scoped version. Not sure what the reason.

File: template/package.json (hunk 5 above)
Comment (hazboun) [reply template/package.json]: yes, react hot loader doesn't work properly. It refreshes the page rather ran updating the components.

File: template/webpack.config.prod.js (hunk 6 above)
Comment (hazboun) [reply template/webpack.config.prod.js:36]: not sure I understand?

File: template/package.json (hunk 3 above)
Comment (markmur) [reply template/package.json:75]: Oh sorry, I meant the `sails-hook-autoreload` dep

File: template/webpack.config.prod.js (hunk 6 above)
Comment (markmur) [reply template/webpack.config.prod.js:36]: lol sorry, I don't know where "extract" came from. I just meant there's unnecessary tabs/spaces before `alias`

File: template/package.json (hunk 5 above)
Comment (markmur) [reply template/package.json]: Hmm, I guess it's a trade off then. Webpack 4 offers enough to make the upgrade so I'd rather go with 4 for now and fix the hot reloading later.

File: template/package.json (hunk 4 above)
Comment (markmur) [reply template/package.json:17]: Is it a required peer dependency?

File: template/package.json (hunk 3 above)
Comment (hazboun) [reply template/package.json:75]: I believe so, yes.

File: template/package.json (hunk 4 above)
Comment (hazboun) [reply template/package.json:17]: yes. The project failed to run without it.

File: template/package.json (hunk 5 above)
Comment (hazboun) [reply template/package.json]: ok, I'll update to latest webpack then

File: template/package.json (hunk 4 above)
Comment (markmur) [reply template/package.json:17]: Ah yes I see it's being required in the config file. That's fine.

File: template/webpack.config.dev.js [hunk 7]
```diff
@@ -44,19 +41,19 @@ module.exports = {
...
+    rules: [
       {
         test: /\.scss$/, // sass files
-        loader: 'style!css!autoprefixer?browsers=last 2 version!sass?outputStyle=expanded',
+        loaders: ['style-loader', 'css-loader?sourceMap', 'sass-loader?sourceMap'],
       },
       {
         test: /\.(ttf|eot|svg|woff)(\?[a-z0-9]+)?$/, // fonts files
-        loader: 'file-loader?name=[path][name].[ext]',
+        loaders: ['file-loader?name=[path][name].[ext]'],
```
Comment (markmur) [inline template/webpack.config.dev.js]: `loaders` will need to be changed to `use` where specified everywhere in this file to comply with `webpack` 4 AFAIK

**Review by markmur — APPROVED** [review]
